### PR TITLE
Move to using DT_INST_<INSTANCE>_<COMPAT>_<PROP>

### PR DESCRIPTION
**Pull request by galak**
There are some possible conflicts with the old `DT_<COMPAT>_<INSTANCE>_<PROP>` defines, so deprecated them in favor of `DT_INST_<INSTANCE>_<COMPAT>_<PROP>`

### CODE REVIEW

**Review by mbolivar — APPROVED** [review]
+1 as long as the old deprecated defines still work  while LTS is around :)

**Comment by galak** [comment]
> +1 as long as the old deprecated defines still work while LTS is around :)

Why?  Are we saying that noting that is marked deprecated can be removed for 2 years?

File: drivers/serial/uart_sifive.c [hunk 1]
```diff
@@ -381,17 +381,17 @@ static void uart_sifive_irq_cfg_func_0(void);
 #endif
 
 static const struct uart_sifive_device_config uart_sifive_dev_cfg_0 = {
-	.port         = DT_SIFIVE_UART0_0_BASE_ADDRESS,
-	.sys_clk_freq = DT_SIFIVE_UART0_0_CLOCK_FREQUENCY,
-	.baud_rate    = DT_SIFIVE_UART0_0_CURRENT_SPEED,
+	.port         = DT_INST_0_SIFIVE_UART0_BASE_ADDRESS,
```
Comment (carlescufi) [inline drivers/serial/uart_sifive.c]: @nategraff-sifive @galak Why is the compat named `UART0`? does it make sense that there is a `0` there?

**Review by carlescufi — APPROVED** [review]
LGTM, just left a question

File: drivers/serial/uart_sifive.c (hunk 1 above)
Comment (galak) [reply drivers/serial/uart_sifive.c]: Fair question, its how @nategraff-sifive and the SiFive set their compat's.  Not directly related to this PR, but worth asking.

**Review by ioannisg — APPROVED** [review]

File: drivers/serial/uart_sifive.c (hunk 1 above)
Comment (nategraff-sifive) [reply drivers/serial/uart_sifive.c]: Yeah, it's just how they are. The idea is that the 0 gets revved to a 1 if/when the interface changes.

**Review by nategraff-sifive — APPROVED** [review]
SiFive drivers LGTM

**Review by sathishkuttan — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Review by vonhust — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]
Looks fine to me

**Review by erwango — COMMENTED** [review]
I've grepped _0_LABEL and looked to things not looking like _0_is part of potential \<device label\>.
I've found following:
```
DT_GPIO_KEYS_BUTTON_0_LABEL (doc/guides/dts/index.rst, to be updated ?)
DT_MICROCHIP_ENC28J60_0_LABEL
DT_TI_TLV320DAC_0_LABEL
DT_NORDIC_NRF_CLOCK_0_LABEL
```
Shouldn't these be updated as well?

**Review by henrikbrixandersen — APPROVED** [review]

**Review by wentongwu — APPROVED** [review]

**Comment by galak** [comment]
> I've grepped _0_LABEL and looked to things not looking like _0_is part of potential <device label>.
> I've found following:
> 
> ```
> DT_GPIO_KEYS_BUTTON_0_LABEL (doc/guides/dts/index.rst, to be updated ?)
> DT_MICROCHIP_ENC28J60_0_LABEL
> DT_TI_TLV320DAC_0_LABEL
> DT_NORDIC_NRF_CLOCK_0_LABEL
> ```
> 
> Shouldn't these be updated as well?

I'll take a look and handle in a follow up PR.

**Comment by galak** [comment]
> I'll take a look and handle in a follow up PR.

Cleaned up some missing conversions in #16831

**Comment by mbolivar** [comment]
> > +1 as long as the old deprecated defines still work while LTS is around :)
> 
> Why? Are we saying that noting that is marked deprecated can be removed for 2 years?

It seems like a small amount of code to maintain, for which a large amount of portability between LTS and later versions is gained. Having had to maintain out of tree applications myself, I think this is something our users would value.

Just my opinion.

**Comment by galak** [comment]
> It seems like a small amount of code to maintain, for which a large amount of portability between LTS and later versions is gained. Having had to maintain out of tree applications myself, I think this is something our users would value.
> 
> Just my opinion.

I understand the possible value towards LTS, and will see how much effort it is to maintain, but not making any claims about keeping it around.

**Comment by mbolivar** [comment]
> I understand the possible value towards LTS, and will see how much effort it is to maintain, but not making any claims about keeping it around.

Understood. I'm making the rounds nagging people to consider portability between old and new applications in general right now :).

**Pull request merged**